examples/mem-performance-test/src/main.rs

```diff
@@ -0,0 +1,270 @@
+// How to run:
+// cargo run --features "mem_basic"   or just: cargo run
+// cargo run --features "mem_hot"
+// cargo run --features "mem_cold"
+// cargo run --features "mem_hot,zero_after_alloc"
+
+#![feature(allocator_api)]
+#![feature(new_uninit)]
+#![feature(vec_push_within_capacity)]
+
+
+use std::alloc::AllocError;
+//use std::env;
+//use crate::error::Error;
+//use std::fmt::Error;
+//use std::error::Error;
+//use std::path::Path;
+use std::time::{Instant, Duration};
+use std::mem;
+// use std::mem::MaybeUninit;
+
+static APP_NAME: &str = "mem-performance-test";
+
+// Test Parameters
+//    println!("    AllocFree: does alloc buf_size followed by free for n_sec");
+//    println!("    AllocManyFreeManyCold: does buf_size allocations for a total of max_alloc ");
+//    println!("      bytes followed of a free for all allocated buffers. after each run the ");
+//    println!("      application exits.");
+//    println!("    AllocManyFreeManyWarm: silmilar to AllocManyFreeManyCold but it ignores the ");
+//    println!("      first run and run each buf_size without restarting the application.");
+
+const POW: u32 = 3; // MIN_SIZE specified as power of 2.  8bytes = 2^3
+const MIN_SIZE: usize = mem::size_of::<u64>();
+// number of seconds to run each buf_size test. Ignored by cold tests.
+const TEST_SEC: u32 = 30;
+// maximum amount of memory to allocate. Specified in GiBi bytes.
+const DEFAULT_MAX_MEM_ALLOC_SIZE: usize = gi_bi_bytes(1);
+
+
+const PAGE_SIZE: usize = 4096;
+const PAGE_STEP_U64: usize = PAGE_SIZE / mem::size_of::<u64>();
+
+// IEC power of two values conversion
+#[inline]
+const fn ki_bi_bytes(val: usize) -> usize { val * 1024 as usize }
+#[inline]
+const fn me_bi_bytes(val: usize) -> usize { ki_bi_bytes (val) * 1024 as usize }
+#[inline]
+const fn gi_bi_bytes(val: usize) -> usize { me_bi_bytes (val) * 1024 as usize }
+
+
+fn mem_alloc(buf_size: usize) -> Result<Box<[u64]>, AllocError> {
+
+    let size = buf_size / mem::size_of::<u64>();
+
+    // #[cfg(never)]
+    #[cfg(not(feature = "zero_after_alloc"))]
```

Comment:
Is this necessary? It can be the default buf when `zero_after_alloc` feature is not used.